draft-ietf-tls-tls13.md

```diff
@@ -2939,6 +2939,12 @@ servers MUST process the client's ClientHello and then immediately
 send the ServerHello, rather than waiting for the client's
 EndOfEarlyData message.
 
+Servers are permitted to buffer and postpone the processing of application
+data sent in 0-RTT until they receive client's Finished. Therefore, when
+using 0-RTT, clients MUST proceed the handshake to the point of sending
+the Finished message even if there are no application data to be sent
+in 1-RTT.
```

Comment:
You are missing the necessary preconditions here: the requirement can only apply after the client has received the entire flight of server messages (ServerHello to server Finished).